Development/nmos/constraints.cpp

```diff
@@ -95,18 +89,28 @@ namespace nmos
             }
             if (constraint.has_field(nmos::fields::constraint_minimum) && subconstraint.has_field(nmos::fields::constraint_minimum))
             {
-                const auto& constraint_minimum = nmos::fields::constraint_minimum(constraint);
-                const auto& subconstraint_minimum = nmos::fields::constraint_minimum(subconstraint);
-                if (constraint_minimum > subconstraint_minimum)
+                if (constraint.at(U("minimum")).has_field(nmos::fields::numerator))
```

Comment:
Would changing the definition of `nmos::fields::constraint_minimum` and `nmos::fields::constraint_maximum` to `web::json:field_as_value` simplify things? Although it would be a breaking change, I don't think many nmos-cpp users will rely on it being `web::json::field<double>`...

Reply:
Do you mean that it would just allow us to get rid of `at` call or I'm missing some additional advantage of it?

Reply:
Yes, maybe that is all, but the inconsistency of use looks a bit strange right now. Actually I think it's a bug that these are defined as `web::json::field<double>` right now because BCP-004-01 parameter constraints have always supported min/max on integer, number and rational. Originally of course these came from IS-05 constraints... but even those support both integer and number, so in some cases `int64_t` would be more appropriate than `double`.

Therefore I think redefining them as `web::json:field_as_value` and then checking `has_field` and/or `is_integer` and `is_double` where necessary would be cleaner?